internal/backoff/exponential.go

```diff
@@ -34,37 +35,47 @@ type ExponentialOption func(*exponentialOptions)
 
 // exponentialOptions are the configuration options for an exponential backoff
 type exponentialOptions struct {
-	base, min, max time.Duration
-	rand           *rand.Rand
-	minMaxDiff     int64
+	first, max time.Duration
+	newRand    func() *rand.Rand
 }
 
 func (e exponentialOptions) validate() (err error) {
-	if e.base <= 0 {
-		err = multierr.Append(err, errors.New("invalid base for exponential backoff, need greater than zero"))
-	}
-	if e.min < 0 {
-		err = multierr.Append(err, errors.New("invalid min for exponential backoff, need greater than or equal to zero"))
+	if e.first <= 0 {
+		err = multierr.Append(err, errors.New("invalid first duration for exponential backoff, need greater than zero"))
```

Comment:
nit (but not really part of this PR): I like most errors created with `errors.New` to be global constants.